src/split-at.mjs

```diff
@@ -0,0 +1,18 @@
+import slice from './slice'
+import tee from './tee'
+import ensureIterable from './internal/ensure-iterable'
+
+function splitAt (position, iter) {
+  const [a0, b0] = tee(ensureIterable(iter))
+  const a1 = slice({ end: position }, a0)
```

Comment:
I would like to see a better implementation. In particular I want to see that if the user consumes the entire first half of the split prior to consuming any of the second, that no cache values are created. If the user begins consuming the second half of the split prior to the first, I want to see and array cache of the correct size created (`new Array(position)`) and filled by calling `next()` `position` times, which must be done by hand in order to avoid closing the source iterable accidentally. This means that you will also need a `try..finally` block to ensure that `iter.return()` is called if `splitAt` returns prior to exhausting `iter`.